test/unittests/t_raii_temp_dir.cc

```diff
@@ -0,0 +1,41 @@
+/**
+ * This file is part of the CernVM File System.
+ */
+
+#include <gtest/gtest.h>
+
+#include <util/pointer.h>
+#include <util/raii_temp_dir.h>
+
+#include "platform.h"
+#include "util/posix.h"
+
+static bool DirExists(const std::string& dir) {
+  platform_stat64 dir_stat;
+  int ret = platform_stat(dir.c_str(), &dir_stat);
+  return ret == 0;
+}
+
+class T_RaiiTempDir : public ::testing::Test {};
+
+TEST_F(T_RaiiTempDir, Basic) {
+  UniquePtr<RaiiTempDir> temp_dir(RaiiTempDir::Create("/tmp/test_dir"));
```

Comment:
The unit tests already have there working directory on a temporary path.  Creation of files and directories should start in `GetCurrentWorkingDirectory()`